subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -56,6 +56,15 @@
 
 #define DISCOVER_PREFACE	"</.well-known/core>;ct=40"
 
+#define RESOURCE_TYPE		";rt=\"oma.lwm2m\""
```

Comment:
Let's have REG_PREFACE = "" when JSON format is not supported and add a comment explaining why we're setting the alternate path when it is supported (one LwM2M server implementation (wakaama) that I used for testing is picky).